File: head-first/chapter01/SimUDuck/src/duck/ModelDuck.java
package duck;

import fly.FlyNoWay;
import quack.Quack;

public class ModelDuck extends  Duck{
    public ModelDuck() {
        flyBehavior = new FlyNoWay();
        quackBehavior = new Quack();
    }

    @Override
    public void display() {
        System.out.println("모형오리");
    }
}

File: head-first/chapter01/SimUDuckKotlin/src/main/kotlin/duck/ModelDuck.kt
package duck

import fly.FlyNoWay
import quack.Quack

class ModelDuck() : Duck() {
    init {
        flyBehavior = FlyNoWay()
        quackBehavior = Quack()
    }

    override fun display() {
        println("모형오리")
    }
}